batching_kafka_consumer/__init__.py

```diff
@@ -235,39 +243,50 @@ def _shutdown(self):
 
     def _reset_batch(self):
         logger.debug("Resetting in-memory batch")
-        self.uncommitted_messages = False
-        self.batch = []
-        self.timer = None
+        self.__batch_results = []
+        self.__batch_deadline = None
+        self.__batch_messages_processed_count = 0
+        self.__batch_messages_processed_duration_ms = 0.0
 
     def _flush(self, force=False):
         """Decides whether the `BatchingKafkaConsumer` should flush because of either
         batch size or time. If so, delegate to the worker, clear the current batch,
         and commit offsets to Kafka."""
-        if self.uncommitted_messages:
-            batch_by_size = len(self.batch) >= self.max_batch_size
-            batch_by_time = self.timer and time.time() > self.timer
-            if (force or batch_by_size or batch_by_time):
-                logger.info(
-                    "Flushing %s items: forced:%s size:%s time:%s",
-                    len(self.batch), force, batch_by_size, batch_by_time
+        if not self.__batch_messages_processed_count > 0:
+            return  # No messages were processed, so there's nothing to do.
```

Comment:
Moved this to an early return, since the nesting level here was getting pretty deep.

Reply:
Do we even need this condition? if we do not pass the one at line 260 we would not pass this one either.

Reply:
Yeah, I think you're right — if there were no messages processed there won't be any in the batch results, and we also won't have set the batch deadline (since the batch deadline is set the first time a message is handled.) On the other hand, this protects against a bunch of divide by zero conditions in the metrics below, and having to wrap those individually seems a bit excessive? I think this might be worth keeping around for safety until the batching logic is refactored, just in case something changes with how the deadline is set.

Reply:
ok